File: Arcade/Intro/07_almostIncreasingSequence.js
function almostIncreasingSequence(sequence) {
    if (sequence.length === 2) {return true;}
    
    var remove = 0, 
        i = 0;
    
    while (remove < 2 && i < sequence.length - 1) {
        
        if (sequence[i] < sequence[i + 1]) {
        } else if (i === 0 && sequence[i] > sequence[i+2]) {
            remove++;
        } else if (i === sequence.length - 2) {
            remove++;
        } else if (i > 0 && sequence[i-1] < sequence[i+1] ) {
            remove++;
        } else if (i < sequence.length - 2 && sequence[i] < sequence[i+2] ) {
            remove++;
        }  else if (i > 1 && sequence[i] < sequence[i-2] ) {
            remove++;
        } else {return false;}
        
        i++;
    } 
    
    return remove > 1 ? false : true;
}

File: Arcade/Intro/C++/07_almostIncreasingSequence.cpp
bool almostIncreasingSequence(std::vector<int> sequence) {
    int removeIndex=0;
    for(;removeIndex<sequence.size()-1;removeIndex++){
        if(sequence[removeIndex]>=sequence[removeIndex+1]) {
            if(removeIndex>=1 && removeIndex<sequence.size()-1 && sequence[removeIndex-1]>=sequence[removeIndex+1]) {
                removeIndex++;
                if(removeIndex>=1 && removeIndex<sequence.size()-1 && sequence[removeIndex-1]>=sequence[removeIndex+1]) return false;
            }
            break;
        }
        if(removeIndex==sequence.size()-1) return true;
    }
    removeIndex++;
    for(;removeIndex<sequence.size()-1;removeIndex++){
            if(sequence[removeIndex]>=sequence[removeIndex+1]) return false;
    }
    return true;
}
